[layout] Create a page size registry for layout

nyalldawson opened the pull request:
## Description

Adds a new QgsPageSize class and QgsPageSizeRegistry registry (attached to QgsApplication), which stores and manages known page sizes

Builds off the classes introduced in #4767

3nids commented:
Due to last changes, you need to discard changes to core.sip, gui.sip and run sip_include.sh

nyalldawson committed:
[layout] Create a page size registry for layout

Adds a new QgsPageSize class and QgsPageSizeRegistry registry
(attached to QgsApplication), which stores and manages known
page sizes
